static/js/components/RunSummary.jsx

```diff
@@ -0,0 +1,428 @@
+      <IconButton
+        aria-controls="simple-menu"
+        aria-haspopup="true"
+        onClick={handleClick}
+        variant="contained"
+      >
+        <BuildIcon />
+      </IconButton>
+      <Menu
+        id="simple-menu"
+        anchorEl={anchorEl}
+        keepMounted
+        open={Boolean(anchorEl)}
+        onClose={handleClose}
+      >
+        {render_items}
+      </Menu>
+    </div>
+  );
+};
+
+SimpleMenu.propTypes = {
+  assignment: PropTypes.shape({
+    status: PropTypes.string,
+    id: PropTypes.number,
+    obj: PropTypes.shape(
+      {
+        id: PropTypes.string
+      }
+    ).isRequired
+  }).isRequired
+};
+
+
+const RunSummary = ({ route }) => {
+  const dispatch = useDispatch();
+  const observingRun = useSelector((state) => state.observingRun);
+  const { instrumentList } = useSelector((state) => state.instruments);
+  const { telescopeList } = useSelector((state) => state.telescopes);
+  const groups = useSelector((state) => state.groups.all);
+
+
+  // Load the observing run and its assignments if needed
+  useEffect(() => {
+    dispatch(Action.fetchObservingRun(route.id));
+  }, [route.id, dispatch]);
+
+
+  if (!(("id" in observingRun) && (observingRun.id === parseInt(route.id, 10)))) {
+    // Don't need to do this for assignments -- we can just let the page be blank for a short time
+    return (
+      <b>
+        Loading run...
+      </b>
+    );
+  } else {
+    const { assignments } = observingRun;
+
+    const renderPullOutRow = ((rowData, rowMeta) => {
+      if (observingRun === undefined) {
+        return "Loading...";
+      }
+
+      const colSpan = rowData.length + 1;
+      const assignment = observingRun.assignments[rowMeta.rowIndex];
+
+      return (
+        <TableRow>
+          <TableCell style={{ paddingBottom: 0, paddingTop: 0 }} colSpan={colSpan}>
+            <Grid
+              container
+              direction="row"
+              spacing={3}
+              justify="center"
+              alignItems="center"
+            >
+              <ThumbnailList
+                thumbnails={assignment.obj.thumbnails}
+                ra={assignment.obj.ra}
+                dec={assignment.obj.dec}
+                useGrid={false}
+              />
+              <Grid item>
+                <Suspense fallback={<div>Loading plot...</div>}>
+                  <AirmassPlot
+                    dataUrl={`/api/internal/plot/airmass/${assignment.id}`}
+                    ephemeris={observingRun.ephemeris}
+                  />
+                </Suspense>
+              </Grid>
+              <Grid item>
+                <Suspense fallback={<div>Loading plot...</div>}>
+                  <VegaPlot
+                    dataUrl={`/api/sources/${assignment.obj.id}/photometry`}
+                  />
+                </Suspense>
+              </Grid>
+            </Grid>
+          </TableCell>
+        </TableRow>
+      );
+    });
+
+
+    const renderObjId = (dataIndex) => {
```

Comment:
Again, it'd be helpful to add comments clarifying that these are just for the MUI datatables `options`/`columns`

Reply:
Fixed